Feature/ep 9 load users

RobertJDevOP opened the pull request:
Se realiza la lista de usuarios retornándolo al template

RobertJDevOP committed:
feat: add users controllers and routes

RobertJDevOP committed:
feat: refresh user model with filters

RobertJDevOP committed:
feat: add IndexRequest Users

RobertJDevOP committed:
refactor: ControllerUsers use Users/IndexRequest

RobertJDevOP committed:
feat: ViewModel for users

RobertJDevOP committed:
feat: model filter logic is added for users

RobertJDevOP committed:
feat: add resource users blade

RobertJDevOP committed:
fix: fix issue of ModelFilters users

RobertJDevOP committed:
style: php cs fixer 1

RobertJDevOP committed:
style: php cs fixer 2

RobertJDevOP committed:
style: php cs fixer 3

RobertJDevOP committed:
style: php cs fixer 4

RobertJDevOP committed:
style: php cs fixer 5

RobertJDevOP committed:
style: php cs fixer 6

RobertJDevOP committed:
feat: add users lang resource

RobertJDevOP committed:
feat: Add filters users model and test

RobertJDevOP committed:
refactor: method where of Abstract class Filter and user condition of enabled_At filter

RobertJDevOP committed:
feature : status is added for the users table

jortiz3109 commented on composer.json:
```diff
@@ -5,7 +5,7 @@
     "license": "MIT",
     "require": {
-        "php": "^7.4|^8.0",
+        "php": "8.0",
```
Ojo con este cambio.. puede derivar en errores luego

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

RobertJDevOP committed:
feature : filter created add and datepicker format buefy

RobertJDevOP committed:
refactor : Accessors are created for the user model applying good practices

RobertJDevOP committed:
refactor : flexbox is added to dynamic filters :recycle:

RobertJDevOP committed:
refactor : delete unnecessary data providers

RobertJDevOP committed:
Delete composer.json

refactor: files are deleted should not go in commits

RobertJDevOP committed:
Delete vendor.js

refactor: files are deleted should not go in commits

RobertJDevOP committed:
Delete DatabaseSeeder.php

refactor: files are deleted should not go in commits

RobertJDevOP committed:
removed merge conflicts

RobertJDevOP committed:
removed merge conflicts

RobertJDevOP committed:
refactor: add filters tests and add changes for Users view model

RobertJDevOP committed:
fix:Continuous integration laravel-tests

RobertJDevOP committed:
style: php cs fixer

RobertJDevOP committed:
style: php cs fixer

RobertJDevOP committed:
fix:Continuous integration laravel-tests

RobertJDevOP committed:
fix:Continuous integration laravel-tests

RobertJDevOP committed:
refactor: models user and cs fixer

RobertJDevOP committed:
refactor:add users to group of routes

RobertJDevOP committed:
fix:combinacion de 8 commits de codecov y git test

debug:laravel-tests fix in git

debug:laravel-tests fix in git

debug:laravel-tests fix in git

fix: fix codecov report

fix: fix codecov report

fix: fix codecov report

fix: fix codecov report

jortiz3109 commented on app/Filters/Conditions/Created_at.php:
```diff
@@ -0,0 +1,15 @@
+use App\Filters\Criteria;
+use Illuminate\Database\Eloquent\Builder;
+
+class Created_at extends Condition
```
El nombre de esta clase no cumple PSR

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

jortiz3109 commented on app/Filters/Conditions/EnabledAt.php:
```diff
@@ -0,0 +1,17 @@
+use App\Filters\Criteria;
+use Illuminate\Database\Eloquent\Builder;
+
+class EnabledAt extends Condition
```
¿Cúal es la función de esta condición?

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

jortiz3109 commented on app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,18 @@
+    {
+        $users = User::filter($request->input('filters', []))->paginate();
+        $viewModel->collection($users);
+        return view('users.index', $viewModel->toArray());
```
Dar un salto de línea para cumplir PSR

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

jortiz3109 commented on app/Models/User.php:
```diff
@@ -14,11 +15,17 @@ class User extends Authenticatable implements MustVerifyEmail
+        'enabled_at',
+    ];
+
+    protected $appends = ['date_formatted',
```
No cumple PSR

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

jortiz3109 reviewed (changes requested):
Falta revisar pruebas

jortiz3109 commented on app/Models/User.php:
```diff
@@ -51,4 +58,14 @@ public function markAsDisabled(): void
+
+    public function getStatusAttribute(): string
+    {
+        return (is_null($this->attributes['enabled_at'])) ? 'Disabled' : 'Enabled';
```
Utilizar traducciones

RobertJDevOP replied:
https://github.com/jortiz3109/escuela-php-panel/commit/a661b2c5bdcd79a029000ff012f6bcb1cbfc2be7

RobertJDevOP committed:
fusion de commits de code review

fix: fix code review :zap:

fix: fix code review :zap:

fix: fix code review :zap:

fix: fix code review :zap:

fix: fix code review :zap:

fix: fix code review :zap:

RobertJDevOP reviewed (commented):
Se hace revisión de lo indicado

asilvabe committed:
Merge branch 'develop' into feature/EP-9_load-users

asilvabe committed:
style: PHP CS FIXER

asilvabe committed:
test: correct namespace

asilvabe committed:
refactor: change filter from enabledAt to status

asilvabe committed:
style: apply PSR-12

asilvabe committed:
feat: organize routes

asilvabe committed:
test(wip): refactor users index tests

JuanZea committed:
feat:Stable stage with error when creating vue components

juancolo committed:
sylte: file naming convention for ViewModel

juancolo committed:
style: references UserIndexViewModel naming convention

juancolo committed:
test: refactor, implement UserDataProviders for filter and index

juancolo committed:
style: php cs fixer were ran

juancolo committed:
test: refactor IndexTest for users

JuanZea committed:
feat: Complete flow of toggling a users state

JuanZea committed:
refactor: Run php fixer

JuanZea committed:
feat: Add test and rename route

JuanZea committed:
Merge branch 'feature/EP-9_load-users' into feature/EP-8-Enable-user-button

JuanZea committed:
refactor: Treating properties independently as separate resources

juancolo committed:
feat: merge ep 28 transaction index

it is merged to implement the html components applied into the referenced branch

juancolo committed:
test: filters from data provider

asilvabe commented on app/Filters/Conditions/Status.php:
```diff
@@ -0,0 +1,19 @@
+        })->when($criteria == 'disabled', function ($q) {
+            $q->whereNull('enabled_at');
+        });
+    }
```
Refactorizar: en vez de usar las cadenas 'enabled' o 'disabled' sino como booleanos

asilvabe commented on app/Filters/Filter.php:
```diff
@@ -49,9 +49,8 @@ protected function where(): self
-            $conditionClass::append($this->query, new Criteria($value));
+            (is_null($value)) ?: $conditionClass::append($this->query, new Criteria($value));
         }
-
```
Actualizar con la rama develop
Se debe revisar el método `conditions` para evitar que se dejen por fuera los falores false, y únicamente sacar los null

asilvabe commented on app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,19 @@
+        $users = User::filter($request->input('filters', []))->paginate();
+        $viewModel->collection($users);
+
+        return view('users.index', $viewModel->toArray());
```
Primero actuaizar con develop.
Si aún se está usando el toArray, eliminarlo.

asilvabe commented on app/Http/Requests/Users/IndexRequest.php:
```diff
@@ -0,0 +1,20 @@
+    {
+        return [
+            'filters' => ['filled', 'array'],
+        ];
```
Revisar qué reglas hacen falta añadir.
Agregar reglas para cada uno de los filtros utilizados.
Recordar usar el `bail`
Revisar también https://github.com/jortiz3109/escuela-php-panel/pull/25#discussion_r774219512

JuanZea committed:
Merge branch 'develop' into feature/EP-8-Enable-user-button

JuanZea committed:
Merge remote-tracking branch 'origin/feature/EP-27_FiltersTests' into feature/EP-8-Enable-user-button

JuanZea committed:
refactor: Test fixes and merges

JuanZea committed:
refactor: Toggle abstracted as trait

JuanZea committed:
refactor: Boolean attribute to status button

JuanZea committed:
fix: Query for email verified time

JuanZea committed:
fix: Users tests and filters

JuanZea committed:
refactor: Sonarcloud bug and fix filter format

JuanZea committed:
refactor: Updating prod assets

JuanZea committed:
fix: Production build issue

juancolo committed:
merge: solve conflicts pull from develop

JuanZea committed:
Merge branch 'develop' into feature/EP-8-Enable-user-button

JuanZea committed:
Merge branch 'feature/EP-9_load-users' into feature/EP-8-Enable-user-button

JuanZea committed:
fix: Source as prod

juancolo committed:
test: use HasAuthenticatedUser

juancolo committed:
fix: url for user index into admin.blade

juancolo committed:
fix: change @lang to trans into index

Commit:
refactor: add controller

Commit:
Merge remote-tracking branch 'origin/feature/EP-8-Enable-user-button' into Feature/EP-51_enable-disable_payment_methods

juancolo committed:
test: fix test for user login

Commit:
refactor: add toggle interface, and refactorize user status

Commit:
style: run fixer

Commit:
test: add test to toggle route

Commit:
refactor: change button sidebar

juancolo committed:
test: fix count test user index filters

juancolo committed:
Merge branch 'develop' of https://github.com/jortiz3109/escuela-php-panel into feature/EP-9_load-users

Commit:
refactor: revert general toggle route

Commit:
feature: add view index

Commit:
test: add test to enabled/disabled user

Commit:
style: run phpcs

Commit:
chore: change title

JuanZea committed:
Merge branch 'feature/EP-9_load-users' into feature/EP-8-Enable-user-button

JuanZea committed:
refactor: Run php fixer

JuanZea committed:
Update StatusTest.php

JuanZea committed:
Merge branch 'develop' into feature/EP-8-Enable-user-button

Commit:
refactor: resolve comments

Commit:
Merge remote-tracking branch 'origin/feature/EP-8-Enable-user-button' into Feature/EP-51_enable-disable_payment_methods

# Conflicts:
#	public/js/app.js
#	public/js/vendor.js
#	public/mix-manifest.json
#	resources/lang/en/common.php
#	resources/views/layouts/admin.blade.php

Commit:
chore: compile assets

Commit:
refactor: merge develop

asilvabe committed:
Merge pull request #37 from jortiz3109/Feature/EP-51_enable-disable_payment_methods

Feature/ep 51 enable disable payment methods

JuanZea committed:
refactor: Update to production files

JuanZea committed:
Merge pull request #25 from jortiz3109/feature/EP-8-Enable-user-button

Feature/EP-8-enable-user-button
